src/main/java/com/transferwise/openbanking/client/oauth/RestOAuthClient.java

```diff
@@ -59,9 +59,11 @@ public AccessTokenResponse getAccessToken(GetAccessTokenRequest getAccessTokenRe
                 AccessTokenResponse.class);
             accessTokenResponse = response.getBody();
         } catch (RestClientResponseException e) {
+            log.error("Call to token endpoint failed, x-fapi-interaction-id [{}]", requestHeaders.getInteractionId(),  e);
```

Comment:
@umair-akram are we happy about putting some logging here or do we want to catch/log/rethrow in the cop service?
also are we allowed to log this `x-fapi-interaction-id`? is it sensitive?

Reply:
With interaction-ID alone we can't identify any user related data so its safe to log

Reply:
this is a public repo. Are you sure this is the only way how you can trace your issue?

Reply:
@ulianaostapenko If I understand your question I guess I could just catch/log/re throw the exception in the cop service where it's using this client...

Reply:
That would be awesome

Reply:
There is nothing wrong in logging the uuid for the request